addons_rksv/pos_rksv/static/src/js/rksv.js

```diff
@@ -110,28 +111,61 @@ odoo.define('pos_rksv.rksv', function (require) {
                         (status.newValue.drivers.rksv.start_receipt_needed === true)) {
                         self.create_start_receipt();
                     }
-                    // Here do check for the year receipt flag - if it is set - then generate the year receipt for this cash register !
-                    if ((self.year_receipt_in_progress === false) &&
+                    if ((self.start_receipt_in_progress === false) &&
                         (self.all_ok()) &&
                         (status.newValue.drivers.rksv) &&
-                        (status.newValue.drivers.rksv.year_receipt_needed) &&
-                        (status.newValue.drivers.rksv.year_receipt_needed === true)) {
-                        self.create_year_receipt();
-                    }
-                    // Here do check for the month receipt flag - if it is set - then generate the month receipt for this cash register !
-                    if ((self.month_receipt_in_progress === false) &&
+                        (status.newValue.drivers.rksv.start_receipt_needed !== undefined) &&
+                        (status.newValue.drivers.rksv.start_receipt_needed === false) &&
+                        (status.newValue.drivers.rksv.has_valid_start_receipt !== undefined) &&
+                        (status.newValue.drivers.rksv.has_valid_start_receipt === false)) {
+                        self.start_receipt_in_progress = true;
+                        self.bmf_register_start_receipt_rpc().then(
+                            function done(response) {
+                                if (response.success == false) {
+                                    self.start_receipt_in_progress = false;
+                                    self.pos.set('cashbox_mode', 'inactive');
+                                    console.log(response.message);
+                                } else {
+                                    self.start_receipt_in_progress = false;
+                                    console.log("Startbeleg wurde erfolgreich eingereicht!");
+                                }
+                            },
+                            function failed(message) {
+                                self.start_receipt_in_progress = false;
+                                console.log(message);
+                            }
+                        )
+                    } 
+                    if (
                         (self.all_ok()) &&
                         (status.newValue.drivers.rksv) &&
-                        (status.newValue.drivers.rksv.month_receipt_needed) &&
-                        (status.newValue.drivers.rksv.month_receipt_needed === true)) {
-                        self.create_month_receipt();
+                        (status.newValue.drivers.rksv.start_receipt_needed !== undefined) &&
+                        (status.newValue.drivers.rksv.start_receipt_needed === false)
```

Comment:
we should check here also for has_valid_start_receipt flag - i think it will else create month receipt as soon as it got a start receipt - and will not wait for the valid start receipt flag